include/xtensor/xfft.hpp

```diff
@@ -0,0 +1,294 @@
+#ifndef XFFT_HPP
+#define XFFT_HPP
+
+#include "xmath.hpp"
+#include "xcomplex.hpp"
+#include "xutils.hpp"
+#include "xaxis_slice_iterator.hpp"
+#include "xio.hpp"
+#include "xview.hpp"
+
+namespace xt 
+{
+    namespace fft
+    {
+        template<typename T = double, typename E1, typename E2>
+        xt::xarray<std::complex<T>> fftconvolve(E1&& xvec, E2&& yvec, std::ptrdiff_t axis = -1);
+        
+        namespace detail
+        {
+            template<typename T>
+            T PI()
```

Comment:
PI is already implementation in numerical constants. Please use that (or was there an issue? In that case please add in comments, such that we can fix it later)

Reply:
Ah ok, you write it below. What was the error?

Reply:
There was a linker error when I tried to use it. I will pass the exact error along in a bit.

Reply:
#2516 Is the issue

Reply:
Left a comment with preferred version of the code when error is resolved.